src/en/developer-layers-interfaces.md

```diff
@@ -141,41 +142,27 @@ We're now ready to implement the provider interface in `provides.py`.
 
 ```python
 from charmhelpers.core import hookenv
-from charms.reactive import hook
-from charms.reactive import RelationBase
-from charms.reactive import scopes
-
-
-class HttpProvides(RelationBase):
-    # Every unit connecting will get the same information
-    scope = scopes.GLOBAL
-
-    # Use some template magic to declare our relation(s)
-    @hook('{provides:http}-relation-{joined,changed}')
-    def changed(self):
-        # Signify that the relationship is now available to our principal layer(s)
-        self.set_state('{relation_name}.available')
-
-    @hook('{provides:http}-relation-{departed}')
-    def departed(self):
-        # Remove the state that our relationship is now available to our principal layer(s)
-        self.remove_state('{relation_name}.available')
-
-    # call this method when passed into methods decorated with
-    # @when('{relation}.available')
-    # to configure the relation data
-    def configure(self, port):
-        relation_info = {
-            'hostname': hookenv.unit_get('private-address'),
-            'port': port,
-        }
-        self.set_remote(**relation_info)
+from charms.reactive import set_flag, clear_flag
+from charms.reactive import Endpoint
+
+
+class HttpProvides(Endpoint):
+    def configure(self, port, hostname=None):
+        """
+        Configure the HTTP relation by providing a port and optional hostname.
+
+        If no hostname is provided, the unit's private-address is used.
+        """
+        for relation in self.relations:
+            relation.send['hostname'] = hostname or hookenv.unit_get('private_address')
```

Comment:
It  might be best to make it explicit that `send` is **the only** way to share anything with the remote unit. All flags etc. are always local, never shared.